File: Solutions/6kyu/6kyu_simple_fun_cards_and_pero.js
function cardsAndPero(s) {

  var seen  = new Set();
  var cards = {'P':13,'K':13,'H':13,'T':13}

  for (var i=0; i < s.length; i += 3) {
    var card = s.slice(i, i+3)
    if (seen.has(card)) { return [-1, -1, -1, -1] }
    seen.add(card)
    cards[card[0]] -= 1
  }  
  return Object.keys(cards).map(k => cards[k])
}

File: Solutions/6kyu/6kyu_simple_fun_cards_and_pero.py
SAME = [-1, -1, -1, -1]

def cards_and_pero(s):
    suits = {'P':13,'K':13,'H':13,'T':13}
    seen  = set()
    for card in (s[i:i+3] for i in range(0, len(s), 3)):
        if card not in seen:
            seen.add(card)
            suits[card[0]] -= 1
        else:
            return SAME
    return [suits[char] for char in 'PKHT']
